### PR TITLE
Always trigger a test run for a missing base to the base code change commit

### PR DESCRIPTION
**Pull request by alexivanov**
This is the default for non-monorepo enabled projects. If we have fallback to an earlier commit enabled we might try to kick off a workflow for an older commit which we can't do so instead of we use the commit from the code change event which is guaranteed to be same or newer.

### CODE REVIEW

File: src/actions/cloud-compute/cloud-compute.ts [hunk 1]
```diff
@@ -81,15 +84,36 @@ export const runMeticulousTestsCloudComputeAction = async (): Promise<void> => {
...
+    logger.info(
+      `Tests already exist for commit ${baseCommitSha} (${baseTestRun.id})`
+    );
+  } else {
+    // We compute and use the base SHA from the code change event rather than the `baseCommitSha` computed above
+    // as `tryTriggerTestsWorkflowOnBase` can only trigger workflow for the HEAD `main` branch commit.
+    // `baseCommitSha` can be an older commit in a monorepo setup (in cases where we selectively run tests for a specific package).
+    // In such cases we won't be able to trigger a workflow for the base commit SHA provided by the backend.
+    // We will instead trigger test run for a newer base which is the base commit SHA from the code change event and
+    // will use that as the base to compare against. This is safe as `codeChangeBase` is guaranteed to be the same
```
Comment (Que3216) [inline src/actions/cloud-compute/cloud-compute.ts:99]: So the bug is?:

 - 1. We bump logic version
 - 2. Github kicks off workflow on temp merge commit for PR (merges main-branch-commit-a to pr-head-commit-b)
 - 3. User pushes new commit to main branch (main-branch-commit-b)
 - 4. Github workflow for PR reaches point where it needs to kick of a re-run on the base commit due to outdated logic version. But now main-branch-commit-a is not head of main branch, and we can only kick of workflow_dispatch on branches rather than specific commits (I guess one hack would be to create a temp branch or tag and then delete it?)

What commit does this use instead of the normal base (presumably temp merge commit base)? I feel like any other commit would generate false diffs?

I guess this comes down to --- how does the behaviour of getBaseAndHeadCommitShas differ from computeTestRunComparisonData?

Comment (Que3216) [reply src/actions/cloud-compute/cloud-compute.ts:99]: Ah, computeTestRunComparisonData takes into account commits to skip over due to paths regex while getBaseAndHeadCommitShas does not?

Comment (alexivanov) [reply src/actions/cloud-compute/cloud-compute.ts:99]: >What commit does this use instead of the normal base (presumably temp merge commit base)? I feel like any other commit would generate false diffs?

We might use an earlier base commit, yes. In monorepo setups we can't guarantee that the we've ran on the latest main which `tryTriggerTestsWorkflowOnBase` was expecting previously. In theory it should be safe to kick off a test run on the latest main commit as if there is a change that would introduce a visual diff we should've run on in the first place. 


Now there are other edge cases which this doesn't solve (might actually be worth documenting too): For example when using HEAD or PR rather than temp. merge commit as head. In this case we need to kick off a non-head of `main` workflow.

**Review by Que3216 — APPROVED** [review]

File: src/actions/cloud-compute/cloud-compute.ts [hunk 2]
```diff
@@ -81,15 +84,36 @@ export const runMeticulousTestsCloudComputeAction = async (): Promise<void> => {
...
+    // or newer commit to `baseCommitSha`.
+    const { base: codeChangeBase } = await getBaseAndHeadCommitShas(event, {
+      useDeploymentUrl: false,
+    });
+    if (codeChangeBase) {
+      const { shaToCompareAgainst: baseShaFromWorkflow } =
+        await tryTriggerTestsWorkflowOnBase({
```
Comment (Que3216) [inline src/actions/cloud-compute/cloud-compute.ts:106]: Maybe make this return a boolean instead?
